dotCMS/src/integration-test/java/com/dotmarketing/cms/urlmap/URLMapAPIImplTest.java

```diff
@@ -90,6 +91,133 @@ public void prepareTest() {
         urlMapAPI = new URLMapAPIImpl();
     }
 
+    /**
+     * UrlMapped Content that lives on the system host should be available on every host where the map
+     * detail page exists (by path). If content does not live on the system host, it should only be
+     * available on the host it is currently one
+     * 
+     * @throws Exception
+     */
+
+
+    @Test
+    public void systemHost_UrlMaps_Should_Work_Across_Hosts() throws Exception {
+
+        final Host host1 = new SiteDataGen().nextPersisted();
+        final Host host2 = new SiteDataGen().nextPersisted();
+        final Host host3 = new SiteDataGen().nextPersisted();
+        
+        final Folder folder1 = new FolderDataGen().name("folder").title("folder").site(host1).nextPersisted();
+        final Folder folder2 = new FolderDataGen().name("folder").title("folder").site(host2).nextPersisted();
+
+
+
+        final Template template1 = new TemplateDataGen().site(host1).nextPersisted();
+        final Template template2 = new TemplateDataGen().site(host2).nextPersisted();
+
+        final HTMLPageAsset pageOnHost1 = new HTMLPageDataGen(folder1, template1).pageURL("my-detail")
```

Comment:
Codacy found an issue: [The String literal "my-detail" appears 4 times in this file; the first occurrence is on line 118](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796110)